src/main/java/seedu/contax/logic/commands/BatchCommand.java

```diff
@@ -0,0 +1,122 @@
+ */
+public class BatchCommand extends Command {
+    public static final String COMMAND_WORD = "batch";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Perform command in batch \n"
+            + "Parameters: "
+            + "COMMAND (must be valid command without index) "
+            + PREFIX_SEARCH_TYPE + "SEARCH_TYPE "
+            + PREFIX_EQUALS + "PATTERN \n"
+            + "Example: " + COMMAND_WORD + " edit "
+            + PREFIX_SEARCH_TYPE + "phone "
+            + PREFIX_EQUALS + "^123 ";
+
+    private final String commandInput;
+    private final SearchType searchType;
+    private final String userInput;
+
+    /**
+     * @param commandInput              details to word of command
+     * @param searchType                search type of field apply matcher
+     * @param userInput                 regex provided by user
+     */
+    public BatchCommand(String commandInput, SearchType searchType, String userInput) {
+        requireNonNull(commandInput);
+        requireNonNull(searchType);
+        requireNonNull(userInput);
+        this.commandInput = commandInput;
+        this.searchType = searchType;
+        this.userInput = userInput.trim();
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Index> indexList = matchInputStringToIndex(model, searchType, userInput);
+        if (indexList.size() == 0) {
+            return new CommandResult(COMMAND_WORD
+                    + ": No result matching \"" + userInput + "\"");
+        }
+        List<CommandResult> commandResultList = new ArrayList<>();
+        for (Index index: indexList) {
+            AddressBookParser addressBookParser = new AddressBookParser();
+            try {
+                String commandText = ParserUtil.parseAndCreateNewCommand(
+                        commandInput, Integer.toString(index.getOneBased()));
+                Command command = addressBookParser.parseCommand(commandText);
+                commandResultList.add(command.execute(model));
+            } catch (ParseException pe) {
+                throw new CommandException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, BatchCommand.MESSAGE_USAGE));
+            }
+        }
+        StringBuilder resultOutput = new StringBuilder();
+        for (CommandResult result: commandResultList) {
+            resultOutput.append(result.getFeedbackToUser()).append("\n");
+        }
+        return new CommandResult(resultOutput.toString());
+    }
+
+    private List<Index> matchInputStringToIndex(Model model, SearchType searchType, String userInput)
+            throws CommandException {
+        List<Index> indexList = new ArrayList<>();
+        List<Person> personList = model.getFilteredPersonList();
+        for (int i = 1; i < personList.size() + 1; i++) {
+            Person person = personList.get(i - 1);
+            Index index = Index.fromOneBased(i);
```

Comment:
This part is unnecessarily complicated.
```suggestion
        for (int i = 0; i < personList.size(); i++) {
            Person person = personList.get(i);
            Index index = Index.fromZeroBased(i);
```